frontend/index.php

```diff
@@ -29,6 +22,40 @@ function time_elapsed_string($datetime, $full = false) {
     return $diff->format('vor %i Minuten');
 }
 
+$htmloutput = '';
+
+foreach ($data as $key => $value) {
+    $isexpired = ($data[$key]['availability']=='0%' ? 'expired' : '');
+
+    $htmloutput.='<div class="deal_badge' .$isexpired.'">
+    <div class="progress" data-label="'.$data[$key]['availability'].'" style="border-color: '.$brandcolors[$key].'; text-shadow: -1px -1px 0 '.$brandcolors[$key].', 0 -1px 0'.$brandcolors[$key].', 1px -1px 0 '.$brandcolors[$key].', 1px 0 0 '.$brandcolors[$key].', 1px 1px 0 '.$brandcolors[$key].', 0 1px 0 '.$brandcolors[$key].', -1px 1px 0 '.$brandcolors[$key].', -1px 0 0'.$brandcolors[$key].';">
```

Comment:
Maybe we should use a templating engine like [twig](https://twig.symfony.com/) for such large text blobs as it becomes unreadable pretty quick

Reply:
In my opinion the only thing that would make this section hard to read is the lack of code hightlighting on the HTML attributes since it is represented as a string. I don't have any experience with templating engines neverthelessI doubt they make things easier on this scale.